engine/src/test/java/io/zeebe/engine/processor/StreamProcessorTest.java

```diff
@@ -545,32 +545,35 @@ public void processRecord(
   }
 
   @Test
-  public void shouldCreateSnapshotOnCloseEvenWhenNoEventProcessed() throws Exception {
+  public void shouldCreateSnapshotOnCloseEvenIfNothingProcessedSinceLastSnapshot()
+      throws Exception {
     // given
-    final CountDownLatch recoveredLatch = new CountDownLatch(1);
-    streamProcessorRule.startTypedStreamProcessor(
-        (processors, context) ->
-            processors.onEvent(
-                ValueType.WORKFLOW_INSTANCE,
-                WorkflowInstanceIntent.ELEMENT_ACTIVATING,
-                new TypedRecordProcessor<UnifiedRecordValue>() {
-                  @Override
-                  public void onRecovered(final ReadonlyProcessingContext context) {
-                    recoveredLatch.countDown();
-                  }
-                }));
-
-    // when
+    final var recoveredLatch = new CountDownLatch(1);
+    final var streamProcessor =
+        streamProcessorRule.startTypedStreamProcessor(
+            (processors, context) ->
+                processors.onEvent(
+                    ValueType.WORKFLOW_INSTANCE,
+                    WorkflowInstanceIntent.ELEMENT_ACTIVATING,
+                    new TypedRecordProcessor<UnifiedRecordValue>() {
+                      @Override
+                      public void onRecovered(final ReadonlyProcessingContext context) {
+                        recoveredLatch.countDown();
+                      }
+                    }));
     final var stateSnapshotController = streamProcessorRule.getStateSnapshotController();
     recoveredLatch.await(5, TimeUnit.SECONDS);
     final var position =
         streamProcessorRule.writeWorkflowInstanceEvent(WorkflowInstanceIntent.ELEMENT_ACTIVATING);
     TestUtil.waitUntil(() -> streamProcessorRule.events().count() >= 1);
     streamProcessorRule.getClock().addTime(SNAPSHOT_INTERVAL);
     TestUtil.waitUntil(() -> stateSnapshotController.getValidSnapshotsCount() == 1);
+
+    // when
     streamProcessorRule.closeStreamProcessor();
 
     // then
+    assertThat(streamProcessor.isClosed()).isTrue();
     final InOrder inOrder = Mockito.inOrder(stateSnapshotController);
     inOrder.verify(stateSnapshotController, TIMEOUT.times(1)).openDb();
     inOrder.verify(stateSnapshotController, TIMEOUT.times(1)).takeSnapshot(position);
```

Comment:
Take snapshot is only called on enforcing right? to be more impl independent we also just could use the waituntil snapshot count is 2

Reply:
Good point - you're right, this relies on the implementation calling this particular method. And now we can get rid of the mock :+1: